### PR TITLE
Application recovery and restart samples

### PR DESCRIPTION
**Pull request by AlinaSiembor**
## Description

Adding Application Recovery and Restart API samples. 
Started from recommended scenario based templates.

## Target Release

1.1 Preview 1.

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [x] Samples build clean with no warnings or errors.
- [x] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).
- [ ] If I am onboarding a new feature, then I must have correctly setup a new CI pipeline for my feature with the correct triggers and path filters laid out in the "Onboarding Samples CI Pipeline for new feature" section in samples-guidelines.md.
- [x] I have commented on my PR `/azp run SamplesCI-<FeatureName>` to have the CI build run on my branch for each of my FeatureName my PR is modifying. This must be done on the latest commit on the PR before merging to ensure the build is up to date and accurate. Warning: the PR will not block automatically if this is not run due to '/azp run' limitation on triggering more than 10 pipelines.

### CODE REVIEW

File: Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/SampleConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace cs_winui_packaged
+{
+    public static class Kernel32
+    {
+        public delegate int RecoveryDelegate(IntPtr parameterData);
+
+        [DllImport("kernel32.dll")]
```
Comment (AdamBraden) [inline Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/SampleConfiguration.cs:16]: Shouldn't you have been able to do all these with CSWin32 and NativeMethods.txt?

Comment (AlinaSiembor) [reply Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/SampleConfiguration.cs:16]: I was actually not familiar with CSwin32 p/invoke generator. Will update to use that.

Comment (AlinaSiembor) [reply Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/SampleConfiguration.cs]: updated

File: Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/cs-winui-packaged.csproj [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <UseWinUI>true</UseWinUI>
+    <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
+    <PublishProfile>Properties\PublishProfiles\win10-$(Platform).pubxml</PublishProfile>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Remove="S3_RebootTypeDetection.xaml" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.0-preview1" />
```
Comment (AdamBraden) [inline Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/cs-winui-packaged.csproj:19]: sorry I missed this earlier - 1.1.0-preview1 is old. either 1.1 or better yet the recently released 1.2.

File: Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/cs-winui-packaged.csproj [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22000.194" />
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.1.588-beta" PrivateAssets="all" />
```
Comment (AdamBraden) [inline Samples/AppLifecycle/RestartRegistration/cs-winui-packaged/cs-winui-packaged.csproj:21]: while you are updating packages, I think this is on 0.2.xxx now as well.

**Review by AdamBraden — APPROVED** [review]
Thanks for adding to the samples repo!

**Pull request merged**